addons/mail/static/src/models/discuss/discuss.js

```diff
@@ -184,6 +184,11 @@ function factory(dependencies) {
          */
         replyToMessage(message) {
             this.update({ replyingToMessage: [['link', message]] });
+            // avoid to reply to a note by a message and vice-versa.
+            // subject to change later by allowing subtype choice.
+            this.replyingToMessageOriginThreadComposer.update({
+                isLog: !message.is_discussion && !message.is_notification
```

Comment:
Not your fault and to improve later, but this feels strange to base everything including the comment on "subtype" but then the condition is on `is_discussion` & `is_notification` (which are just formatter result based on subtype anyway).

Reply:
task-2291578

That being said I think this PR should be good enough.

Reply:
Don't think it's worth mentioning this task here. More like a TODO on these fields, so not reference of task in code is good.